pragma-node/src/infra/repositories/entry_repository.rs

```diff
@@ -708,3 +705,225 @@ pub async fn get_ohlc(
+            .len();
+        if num_unique_publishers < MINIMUM_NUMBER_OF_PUBLISHERS {
+            return None;
+        }
+
+        median_entries.push(median_entry);
+    }
+    if found_pairs.len() == pairs_set.len() {
+        Some(median_entries)
+    } else {
+        None
+    }
+}
+
+/// Builds a SQL query that will fetch the recent prices between now and
+/// the given interval for each unique tuple (pair_id, publisher, source)
+/// and then calculate the median price for each pair_id.
+/// We also return in a JSON string the components that were used to calculate
+/// the median price.
+fn build_sql_query_for_median_with_components(pair_ids: &[String], interval_in_ms: u64) -> String {
+    format!(
+        r#"
+            WITH last_prices AS (
+                SELECT
+                    e.pair_id,
+                    e.publisher,
+                    p.account_address AS publisher_account_address,
+                    e.source,
+                    e.price,
+                    e.timestamp,
+                    e.publisher_signature,
+                    ROW_NUMBER() OVER (PARTITION BY e.pair_id, e.publisher, e.source ORDER BY e.timestamp DESC) AS rn
+                FROM 
+                    entries e
+                JOIN
+                    publishers p ON e.publisher = p.name
+                WHERE 
+                    e.pair_id IN ({pairs_list})
+                    AND e.timestamp >= NOW() - INTERVAL '{interval_in_ms} milliseconds'
+            ),
+            filtered_last_prices AS (
+                SELECT 
+                    pair_id,
+                    publisher,
+                    publisher_account_address,
+                    source,
+                    price,
+                    timestamp,
+                    publisher_signature
+                FROM 
+                    last_prices
+                WHERE 
+                    rn = 1
+            )
+            SELECT
+                pair_id,
+                percentile_cont(0.5) WITHIN GROUP (ORDER BY price) AS median_price,
+                jsonb_agg(
+			        jsonb_build_object(
+			            'pair_id', pair_id,
+			            'price', price,
+			            'timestamp', timestamp,
+			            'publisher', publisher,
+                        'publisher_address', publisher_account_address,
+			            'publisher_signature', publisher_signature
+			        )
+			    ) AS components
+            FROM
+                filtered_last_prices
+            GROUP BY 
+                pair_id;
+            "#,
+        pairs_list = pair_ids
+            .iter()
+            .map(|pair_id| format!("'{}'", pair_id))
+            .collect::<Vec<String>>()
+            .join(", "),
+        interval_in_ms = interval_in_ms
+    )
+}
+
+// TODO(akhercha): sort this out - do we want a limit ?
+// TODO(akhercha): What happens then if we still have nothing? Currently we raise error 404 & break the channel.
+pub const LIMIT_INTERVAL_IN_MS: u64 = 10000;
```

Comment:
move to `constants.rs`

Reply:
I like the idea of having a constant.rs file for general things like `PRAGMA_ORACLE_NAME_IN_STARKEX` but do we want them for those kind of constants too?

I think it's clearer to have them very close to the function they're used in, make it clearer then to have every constants in one module even though they don't really share the same logic